subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -144,44 +145,17 @@ private void configureJacocoReportsDefaults(final JacocoPluginExtension extensio
     }
 
     private void configureJacocoReportDefaults(final JacocoPluginExtension extension, final JacocoReport reportTask) {
-        reportTask.getReports().all(new ReportEnabledAction(project.getProviders()));
-        reportTask.getReports().all(new ReportOutputDirectoryAction(project.getProviders(), reportTask, project.provider(extension::getReportsDir)));
-    }
-
-    private static class ReportEnabledAction implements Action<ConfigurableReport> {
-
-        private final ProviderFactory providers;
-
-        private ReportEnabledAction(ProviderFactory providers) {
-            this.providers = providers;
-        }
-
-        @Override
-        public void execute(ConfigurableReport report) {
-            report.setEnabled(providers.provider(() -> report.getName().equals("html")));
-        }
-    }
-
-    private static class ReportOutputDirectoryAction implements Action<ConfigurableReport> {
-
-        private final ProviderFactory providers;
-        private final JacocoReport reportTask;
-        private final Provider<File> reportsDir;
-
-        private ReportOutputDirectoryAction(ProviderFactory providers, JacocoReport reportTask, Provider<File> reportsDir) {
-            this.providers = providers;
-            this.reportTask = reportTask;
-            this.reportsDir = reportsDir;
-        }
-
-        @Override
-        public void execute(ConfigurableReport report) {
+        reportTask.getReports().all((Action<ConfigurableReport> & Serializable) report ->
```

Comment:
💄 `action(report -> ...)` where `action` is `SerializableLambdas.action`.

Reply:
Oh I missed that `action(..)` helper, nice! 👍